### DISCUSSION TITLE
Transferring metadata from convertion of image.fpx to image.jpg

### DISCUSSION THREAD

**Discussion by Budgie2** [ImageMagick/ImageMagick#2827] [General]
When I ran convert for a batch of .fpx images the metadata was not then available from the .jpg image.
The files had been downloaded from camera and just given sequential number.  I can see some metadata, in particular the date when picture was taken when I look at the thumbnail but these data does not appear when I open the jpeg image using  digiKam.  Am I missing a command option?

I tried to post on the old Users forum but that has moved and login no longer works.  Perhaps my registration has been deleted; whatever I cannot find a way to register again so am using Git discussion meanwhile.

Grateful for any guidance please.

**Comment by Budgie2** [comment]
Further to this first my thanks for the work on libfpx which is now working well in my installation.  All good here but I still have a problem when using IM as an image loader plugin in digikam.  Unfortunately fpx files crash digikam.  I have posted a bug for digikam and do not believe there is a bug with IM but had hoped to be able to extract the metadata and particularly the date when the picture was taken.  How may I extract this from original fpx file using IM please.

**Comment by snibgo** [comment]
What version of IM, on what platform?

I know nothing about fpx and what metadata IM can read from them, but this should tell you:
```
magick myfile.fpx -verbose info:
```
The text output contains all the metadata IM can read. If the metadata you want isn't there, then IM can't read it.

IM is a raster image processor. In general, it reads all metadata needed to interpret the image, but doesn't bother about metadata that is not needed. exiftool is better for that. So you might use IM to process the image, and exiftool to copy metadata from the input fpx to the output jpeg.

**Reply by Budgie2** [reply]
Since my last post I have done some more work going back to the original copy fpx files I have.  I have used the command:-
`magick mogrify -format jpg *.fpx
`
This gives me a virtually identical picture resolution but no metadata has been copied over.  I shall need an explicit additional command to ensure the metadata is copied over and that is beyond me.I have checked the original fpx images using the above command `magick myfile.fpx -verbose info:` and they do have the date I want although shown as date:modify: NOT date:create: so I need to ensure this data is written into the converted images with the required date which is when the picture was taken.  I hope I can get some help here as I cannot get much further on my own.

**Comment by Budgie2** [comment]
Hi,
Many thanks.  Your command gave me what I needed.

Interesting but what I recognise to be the date when picture was taken is shown as 
date:modify 1998-12-12T15:35:22+00:00

and my recent moving the file is shown as
date:create 2020-11-13T14:36:00+00:00

I shall need to remember this.  Many thanks,
Budge

**Reply by Budgie2** [reply]
OK I have the information and it is clear that some of my pictures have been converted from fpx to jpg using different applications and so the same picture when converted has different resolutions and also the metadata is not included.  My best solution is to re convert from my original data.  I therefore wish to convert a directory full of subdirectories of pictures and convert them all to jpg with the best resolution possible from the data and include all the metadata in the new image.
I have read the relevant manual but need some help please with how to define "best" and add metadata.  Still a beginner with command line so I would  really appreciate some help.
Many thanks,
Budge

**Comment by snibgo** [comment]
> ... and they do have the date I want although shown as date:modify: NOT date:create: so I need to ensure this data is written into the converted images with the required date ...

I think "date:modify" and "date:create" are metadata about the files from the file system (part of the operating system), rather than from metadata contained within the files.

If I am correct, then you need to use O/S facilities to copy these dates. Sorry, I don't know how.

Metadata that is contained within files can be copied by exiftool:
```
exiftool -tagsfromfile in.fpx out.jpg
```

**Reply by Budgie2** [reply]
Hi snibgo,
Hopelessly lost and this is the only way I can get through.  No registration possible on forum it appears.
I need to raise a new question on this thread.  I have built IM again with delegate fpx library installed.  I thought jpeg was a standard included module but on checking my configuration jpeg is not included.  Do I have to download build and install a module for jpeg now?

**Reply by fmw42** [reply]
What version of IM and what platform? How did you install IM? From source or binary?

**Reply by Budgie2** [reply]
I downloaded ImageMagick.tar.gz from the link in the Install from Linux Source section which gives me version 7.0.10-62.
I am running on openSUSE Tumbleweed.
I was able to build without any problem on second attempt and fpx has been enabled but it appears not jpg!

**Comment by Budgie2** [comment]
On 21/11/2020 21:08, snibgo wrote:
>     ... and they do have the date I want although shown as date:modify: NOT date:create: so I need to ensure this data is written into the converted images with the required date ...
> 
> I think "date:modify" and "date:create" are metadata about the files from the file system (part of the operating system), rather than from metadata contained within the files.
> 
> If I am correct, then you need to use O/S facilities to copy these dates. Sorry, I don't know how.
> 
> Metadata that is contained within files can be copied by exiftool:
> 
> |exiftool -tagsfromfile in.fpx out.jpg |
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/ImageMagick/ImageMagick/discussions/2827#discussioncomment-135509>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AALT4X6QNKD3PMZUIWBNPFLSRAT5HANCNFSM4TVTZPLQ>.
> 

Hi snibgo and thanks for the reply.  I am bottom posting and still getting used to this list.
I understand your point about the O/S and it is true that previous copying has confused the dates but the original fpx picture does have the date and Imagemagick can see it.  

I have therefore tried using the exiftool on a pair of images as you suggest and I now have a jpg thumbnail with the correct date and when I look at metadata in digikam all I need is present, so many thanks for this.  All I need to do now is try and get my batch processing correct!
Regards,
Budge
